### ISSUE TITLE
[Bug] nil pointer dereference

### ISSUE THREAD

**Issue by jamiek23** [grafana/grafana#6731]
In Grafana 4.0.0-beta2, if the config isn't loaded for whatever reason it crashes without a helpful error message. It is very similar to #6636, but on a slightly different line. Environment is Debian 8/Jessie, as opposed to Ubuntu 14.04. Was installed via apt/dpkg - grafana_4.0.0-1479719016beta2_amd64.deb.

The issue seems to be at line 328 of setting.go: https://github.com/grafana/grafana/blob/b2edcd8cfde2caa0484fc2887c03015cccd6f5eb/pkg/setting/setting.go#L328

Output from grafana is below:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4f09fd]
goroutine 1 [running]:
panic(0xd23800, 0xc420010070)
/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/grafana/grafana/pkg/setting.loadSpecifedConfigFile(0x7ffe994c0d83, 0x18, 0x3, 0xc4201ca420)
/go/src/github.com/grafana/grafana/pkg/setting/setting.go:327 +0xed
github.com/grafana/grafana/pkg/setting.loadConfiguration(0xc4200fdde8)
/go/src/github.com/grafana/grafana/pkg/setting/setting.go:374 +0x1ee
github.com/grafana/grafana/pkg/setting.NewConfigContext(0xc4200fdde8, 0x1, 0xc4200fddf8)
/go/src/github.com/grafana/grafana/pkg/setting/setting.go:460 +0x54
main.initRuntime()
/go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/main.go:69 +0xd5
main.(*GrafanaServerImpl).Start(0xc4201a27e0)
/go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:52 +0x5f
main.main()
/go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/main.go:61 +0x12c
```

**Comment by torkelo** [comment]
could you show your config file? Do you start grafana in any particular way?

**Comment by jamiek23** [comment]
To save you a bit of time, I've found the issue with my config file is on the last line - an un-terminated square bracket. I've pasted my broken config here, with domains and secrets substituted: https://gist.github.com/jamiek23/c7d53814b03070b29afb530c0369da3d

Grafana is started using the normal init.d/systemctl commands, which produced the following output...
`/usr/sbin/grafana-server --config=/etc/grafana/grafana.ini --pidfile=cfg:default.paths.logs=/var/log/grafana cfg:default.paths.data=/var/lib/grafana cfg:default.paths.data=/var/lib/grafana cfg:default.paths.plugins=/var/lib/grafana/plugins`

Don't know if a go panic() is the best way to handle broken user input? :smile:

**Comment by bergquist** [comment]
No panic is a terrible way to handle user input :)

Thank you for reporting this! Fixed in master and will be available in 4.0.0

**Comment by jamiek23** [comment]
Thanks @bergquist :+1:
